pkg/driver/driver.go

```diff
@@ -35,6 +35,14 @@ type Driver interface {
 	// It returns error if there are any errors during Stop
 	Stop(_ context.Context) error
 
+	// Register will add an instance to a registry.
+	// It returns error if there are any errors during Register
+	Register(_ context.Context) error
```

Comment:
Does create and delete instead of Register and Unregister makes more sense ?

We call unregister/delete -> during delete of instance
We call register/create -> during either create/start of instance

Reply:
Good point, just moved it to `createInstance`

Reply:
The VM was already created or deleted, the API call just removed it from the global definition (in VirtualBox)

If you didn't do this, it would still be listed in the GUI - just gave a "file not found" when you tried to access it.

We actually never called `Register`, it was easier to use the `-register` parameter.

The main reason to add it to the API, was so that the driver could decide what to do.

Reply:
For most drivers these would be no-op. But the "hook" was needed, for the vbox driver.